Improve install role

l00ptr opened the pull request.

l00ptr committed:
Optimize apt / dnf package install

We call package manager only once by comibining list of package we want
or need to install.

l00ptr committed:
Reorganize set-domain tasks with block

l00ptr committed:
Add molecule scenario for install with Apache

l00ptr committed:
Add yamlint configuration

l00ptr committed:
Apply yamlfixer to avoid yamllint error

l00ptr committed:
Improve install role meta info

l00ptr committed:
Remove unused test dir on install role

l00ptr commented:
any feedback about that MR ?

l00ptr committed:
Use FQCN for module name (Ansible best practice)

We use the fqcn when calling all Ansible module within our tasks.

l00ptr committed:
Fix gpg key for debian and redhat

coudot commented:
Hello @l00ptr 

thanks for the contributions. I do not use the ansible role, I don't have any feedback to give.

For the moment the Molecule test seems to fail, you should fix it before we merge.

xavierba commented:
This will need to be rebased once #16, which addresses only linting, has been merged.

xavierba commented on roles/install/tasks/debian.yml:
```diff
@@ -13,7 +13,7 @@
 
 - name: Import LLNG GPG KEY
-  ansible.builint.apt_key:
+  ansible.builtint.apt_key:
```
there's still a typo here: builtint --> builtin
also, please squash with previous commit

l00ptr replied:
fixed

xavierba commented on roles/install/vars/main.yml:
```diff
@@ -0,0 +1,31 @@
+__redhat_apache_pkgs:
+  - httpd
+  - mod_perl
+  - mod_fcgid
```
mod_perl and mod_fcgid should be pulled in by the rpm packages dependencies.

l00ptr replied:
removed!

xavierba commented on roles/install/vars/main.yml:
```diff
@@ -0,0 +1,31 @@
+__redhat_nginx_pkgs:
+  - nginx
+  - lemonldap-ng-fastcgi-server
+  - lemonldap-ng-nginx
```
Starting with llng 2.19.0, -nginx subpackage doesn't exist anymore, the nginx configurations have been move to their respective sub-packages.

xavierba commented on roles/install/vars/main.yml:
```diff
@@ -0,0 +1,31 @@
+  - lemonldap-ng-fastcgi-server
+  - lemonldap-ng-nginx
+__redhat_default_pkgs:
+  - perl-LWP-Protocol-https
```
This is missing in the RPMs Requires, I will fix that, thanks :-)

l00ptr replied:
so let me know when it's done on the RPM side and i will also remove that package

xavierba replied:
The dependency has been explicitly added with https://gitlab.ow2.org/lemonldap-ng/lemonldap-ng/-/merge_requests/513

But anyway, this was already an indirect dependency from perl-Soap-Lite and perl-Sentry-Raven.

l00ptr replied:
so i removed that part!

l00ptr committed:
Replace dnf by more generic package module

l00ptr commented:
@xavierba [Ansible 2.17 do not work properly with python 3.6](https://github.com/ansible/ansible/issues/82068) (which is the default on rockylinux8); we probably need to install python39 (or greater) or officially don't support rockylinux8 with our collection... any idea about that specific topic ?

xavierba commented on roles/install/tasks/set-domain.yml:
```diff
@@ -1,51 +1,55 @@
-    - Reload Webserver
-    - Enable and reload lemonldap
+- name: Reconfigure lemonldap (based on search)
+  block:
```
there's only one module call under this 'block', so 'block' is useless.

xavierba commented on roles/install/tasks/set-domain.yml:
```diff
@@ -57,27 +61,29 @@
+      with_items:
+        - headers
+        - rewrite
+      when:
```
'when' is duplicated and thus can be moved at the 'block' level.

l00ptr committed:
Use ansible-core 2.16 in CI till better for RHEL8

Default python install on RHEL8 don't work well with Ansible core 2.17,
so we use explicitly ansible-core 2.16 in CI for our tests.

xavierba commented:
Looks good to me, thank you @l00ptr !

xavierba commented:
Just one thing, could you please squash the various fixup commits with the commit they fix ?
I'll merge the PR once done.

l00ptr commented:
> Just one thing, could you please squash the various fixup commits with the commit they fix ? I'll merge the PR once done.

!done

xavierba commented on roles/install/tests/test.yml:
This file deletion probably belongs to " Remove unused test dir on install role" commit

xavierba reviewed (commented):
Other than the file deletion in an unrelated commit, LGTM
